.github/workflows/publish-on-pypi.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish release to PyPI
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+
+jobs:
+  build-and-upload:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out the release commit
+      uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+    - name: Install Python packages needed for build and upload
+      run: python -m pip install build twine
```

Comment:
Do you want to do a `python -m pip install --upgrade pip` first?

Reply:
I don't; it's not useful. :-) I only discovered today that the `setup-python` workflow _already_ updates `pip` to the latest version, so all those `python -m pip install --upgrade pip` commands I've been sprinkling around are useless.

It's a _little_ bit useful in that I often copy the commands from a CI workflow to a prompt when attempting to replicate that CI workflow, and of course the pip update _is_ useful locally. But that doesn't seem like sufficient reason to deliberately include an instruction that won't do anything.

(My thought processes earlier today started with: "Why doesn't setup-python automatically update pip? Maybe I should open an issue." and ended with "Oh. It does.")